### PR TITLE
Update hdmedians to >= 0.14.1

### PR DESCRIPTION
**Pull request by danodonovan**
Following from #1690 this updates the `hdmedians` dependency to avoid an issue where numpy had to be manually installed. This should fix the rest of #1671 

Please complete the following checklist:

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/COPYING.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+
+### Bug fixes
+
+### Deprecated functionality [stable]
+
+### Deprecated functionality [experimental]
+
+### Miscellaneous
+
+* Update hdmedians package to a version which doesn't require an initial manual numpy install.
```
Comment (ElDeveloper) [inline CHANGELOG.md:21]: Would you mind including this in the 0.5.7 section? At some point we mixed the 0.5.6 notes with the 0.5.7 and now everything is scrambled. If you can put this in 0.5.7 I'll submit a later PR and clean this up.

**Review by ElDeveloper — CHANGES_REQUESTED** [review]
Looks good, just one comment.

File: CHANGELOG.md (hunk 1 above)
Comment (danodonovan) [reply CHANGELOG.md]: I hope 8a72c7a covers this - please ping me if I've got myself muddled :-)

**Comment by ElDeveloper** [comment]
thanks @danodonovan

**Pull request merged**